driver/csiplugin/gpfs_util.go

```diff
@@ -381,3 +430,76 @@ func numberInSlice(a int, list []int) bool {
 	}
 	return false
 }
+
+func getVolIDMembers(vID string) (scaleVolId, error) {
+	splitVid := strings.Split(vID, ";")
+	var vIdMem scaleVolId
+
+	if len(splitVid) == 3 {
+		/* This is LW volume */
+		/* <cluster_id>;<filesystem_uuid>;path=<symlink_path> */
+		vIdMem.ClusterId = splitVid[0]
+		vIdMem.FsUUID = splitVid[1]
+		SlnkPart := splitVid[2]
+		slnkSplit := strings.Split(SlnkPart, "=")
+		if len(slnkSplit) < 2 {
+			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vID))
+		}
+		vIdMem.SymLnkPath = slnkSplit[1]
+		vIdMem.IsFilesetBased = false
+		return vIdMem, nil
+	}
+
+	if len(splitVid) == 4 {
+		/* This is fileset Based volume */
+		/* <cluster_id>;<filesystem_uuid>;fileset=<fileset_id>;path=<symlink_path> */
+		vIdMem.ClusterId = splitVid[0]
+		vIdMem.FsUUID = splitVid[1]
+		fileSetPart := splitVid[2]
+		fileSetSplit := strings.Split(fileSetPart, "=")
+		if len(fileSetSplit) < 2 {
+			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vID))
+		}
+
+		if fileSetSplit[0] == "filesetName" {
+			vIdMem.FsetName = fileSetSplit[1]
+		} else {
+			vIdMem.FsetId = fileSetSplit[1]
+		}
+
+		SlnkPart := splitVid[3]
+		slnkSplit := strings.Split(SlnkPart, "=")
+		if len(slnkSplit) < 2 {
+			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vID))
+		}
+		vIdMem.SymLnkPath = slnkSplit[1]
+		vIdMem.IsFilesetBased = true
+		return vIdMem, nil
+	}
+
+	if len(splitVid) == 8 {
+		/* Volume ID created from 2.5.0 onwards  */
```

Comment:
It should be:
/* Volume ID created from 2.5.0 onwards  for new storageclass type*/

Reply:
From 2.5.0 onwards, volume ID will be of the same format for volumes created with new as well as old storage class

Reply:
This will be the format of newly created volume IDs from 2.5.0 onwards: 
<storageclass_type>;<type_of_volume>;<cluster_id>;<filesystem_uuid>;<consistency_group>;<application>;<fileset_name>;<symlink_path>

But we still support parsing the volume IDs of old formats (for processing volumes created before 2.5.0 and used after upgrade to 2.5.0 or later).

Reply:
I see